tests/deployment/test_steps.py

```diff
@@ -1177,3 +1177,20 @@ async def test_pip_install_reqs_with_directory_step_output_succeeds(
             stderr=ANY,
             stdout=ANY,
         )
+
+    async def test_pip_install_fails_on_error(self, monkeypatch):
```

Comment:
I'm not sure that this test needs the `monkeypatch`, is it doing something to make `run_step` work correctly?

Reply:
nope! I am just a lazy copy pasta-er 😬